src/odemis/acq/align/autofocus.py

```diff
@@ -120,21 +121,31 @@ def Measure1d(image):
     Given an image of a 1 line ccd, measure the focus based on the inverse of the width of a gaussian fit of the data.
     It is assumed that the signal is in focus when the width of the signal is smallest and therefore sigma is smallest.
     image (model.DataArray): 1D image from 1 line ccd.
-    returns (float): The focus level of the image.
+    returns (float): The focus level of the image, based on the inverse of the width of a gaussian fitted on the image.
     """
-    def gauss(x, *p):
-        a, b, c, d = p
-        y = a * numpy.exp(-numpy.power((x - b), 2.) / (2. * c ** 2.)) + d
+    # Use the gauss function to fit a gaussian to the 1 line image.
+    def gauss(x, amplitude, pos, width, base):
+        y = amplitude * numpy.exp(-(x - pos) ** 2 / (2 * width ** 2)) + base
         return y
     # squeeze to make sure the image array is 1d.
     signal = numpy.squeeze(image)
+    # Apply a median filter with a kernel of 5, to filter out (at the most 2 neighbouring) outliers.
```

Comment:
This comments explains exactly what the code below does. Please explain _why_ it does it. IE, to handle noise with up to 2 pixels with very large values, that would look like a peak, which sometimes happens on CCDs.